File: binary_search.py
# User define 
def binary_search(data, size, key):
    left = 0
    right = size -1 
    while left <= right:
        mid_index = (left + right)//2
        mid_val = data[mid_index]
        if key == mid_val:
            return mid_index
        elif key > mid_val:
            left = mid_index + 1
        else:
            right = mid_index - 1

    return -1

data = [-5, 5, 10, 15, 20, 50, 100, 150, 200, 300]
print(binary_search(data, len(data), 5))

File: c or c++/binary_search.c
#include<stdio.h>

int main(){
    int temp;
    int len=5;
    int arr[5]={2,3,4,5,6};
    int key=3;
    int left = 0;
    int right = len-1;
    while (left<=right){
        int mid_ind = (left + right)/2;
        int mid_val = arr[mid_ind];
        if (key == mid_val){
            printf("%d\n", mid_ind);
            temp = arr[mid_ind];
            break;

        }
        else if (key > mid_val){
            left = mid_ind + 1;
        }
        else{
            right = mid_ind - 1;
        }

    }
    if (key != temp){
        int var = -1;
        printf("%d", var);

    }

    
    return 0;
}